client.go

```diff
@@ -433,7 +433,7 @@ func NewClient[TTx any](driver riverdriver.Driver[TTx], config *Config) (*Client
 	// There are a number of internal components that are only needed/desired if
 	// we're actually going to be working jobs (as opposed to just enqueueing
 	// them):
-	if config.willExecuteJobs() {
```

Comment:
Should we also make it a config error to specify `Queues` with no pool? `Queues` implies there will be workers working on those queues, which doesn't make sense if those workers could never be enabled.

Reply:
Hah, so that occurred to me as well, but I was 50/50 on the best option. The slight downside of erroring on `NewClient` is that it makes it slightly harder to share one test client between everything (e.g. in our suite, sending a `nil` database pool driver combined with a config from our general `newTestConfig` would be an error because the latter includes `Queues`.

Let's try it out though. Pushed.